source/common/network/apple_dns_impl.cc

```diff
@@ -87,22 +87,55 @@ ActiveDnsQuery* AppleDnsResolverImpl::resolve(const std::string& dns_name,
                                               DnsLookupFamily dns_lookup_family,
                                               ResolveCb callback) {
   ENVOY_LOG(debug, "DNS resolver resolve={}", dns_name);
-  std::unique_ptr<PendingResolution> pending_resolution(
-      new PendingResolution(*this, callback, dispatcher_, main_sd_ref_, dns_name));
 
-  DNSServiceErrorType error = pending_resolution->dnsServiceGetAddrInfo(dns_lookup_family);
-  if (error != kDNSServiceErr_NoError) {
-    ENVOY_LOG(warn, "DNS resolver error ({}) in dnsServiceGetAddrInfo for {}", error, dns_name);
-    return nullptr;
+  Address::InstanceConstSharedPtr address{};
+  try {
+    // After production testing with this resolver implementation it became empirically demonstrable
+    // that Apple's API issues queries to the DNS server for dns_name that might already be an IP
+    // address. In contrast, c-ares synchronously resolves such cases. Moreover, some DNS servers
+    // might not resolve IP addresses and thus result in callback targets that never get a
```

Comment:
I think phrasing it this way is a little bit backwards. Even if we have an example of a DNS server that does "resolve" an IP address, I wouldn't bet on this being standard or expected. I think this comment can probably be truncated to something along the lines of, "When an IP address is submitted to c-ares it synchronously returns it without submitting a DNS query. Because we've come to rely on this behavior, this resolver implements a similar resolution path to avoid making improper DNS queries for resolved IPs."